mart/attack/perturber.py

```diff
@@ -27,73 +29,139 @@
 class Perturber(pl.LightningModule):
     """Peturbation optimization module."""
 
+    MODALITY_DEFAULT = "default"
+
     def __init__(
         self,
         *,
-        initializer: Initializer,
         optimizer: Callable,
-        composer: Composer,
         gain: Gain,
-        gradient_modifier: GradientModifier | None = None,
-        projector: Projector | None = None,
+        composer: Composer | dict[str, Composer],
+        initializer: Initializer | dict[str, Initializer],
+        gradient_modifier: GradientModifier | dict[str, GradientModifier] | None = None,
+        projector: Projector | dict[str, Projector] | None = None,
         objective: Objective | None = None,
+        optim_params: dict[str, dict[str, Any]] | None = None,
```

Comment:
Should we just make optimizer a `Callable | dict[str, Callable]`? That will be more uniform like the existing changes.

Reply:
It will probably simplify a lot of the code changes to the parameters.